pkg/rhos/ocpgwdeployer.go

```diff
@@ -0,0 +1,173 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package rhos
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack"
+	"github.com/pkg/errors"
+	"github.com/submariner-io/cloud-prepare/pkg/api"
+)
+
+type ocpGatewayDeployer struct {
+	CloudInfo
+	projectID string
+}
+
+// NewOcpGatewayDeployer returns a GatewayDeployer capable of deploying gateways using OCP
+// If the supplied cloud is not a RHOS, an error is returned.
```

Comment:
It says that if the supplied cloud is not RHOS, an error is returned, but there is no such check.